### PR TITLE
Dynamic configuration API documentation refactor

### PR DESCRIPTION
**Pull request by ektravel**
### Reorganization of the Dynamic configuration API page
This PR:

Adds headings to the page to separate methods
Adds curl and HTTP code tabs
Adds request and response examples
Improves descriptions of methods
Updates links

#### Preview
https://dynamic-configuration-o99lkupvb-katyamacedo-bigcommerce.vercel.app/docs/latest/api-reference/dynamic-configuration-api

This PR has:

- [x] been self-reviewed

### CODE REVIEW

File: docs/api-reference/dynamic-configuration-api.md [hunk 1]
```diff
@@ -3,73 +3,663 @@ id: dynamic-configuration-api
...
 
 ## Coordinator dynamic configuration
 
-See [Coordinator Dynamic Configuration](../configuration/index.md#dynamic-configuration) for details.
+The Coordinator has dynamic configurations to tune certain behavior on the fly, without requiring a service restart.
+For information on the supported properties, see [Coordinator dynamic configuration](../configuration/index.md#dynamic-configuration).
+
+### Get dynamic configuration
+
+Retrieves the current Coordinator dynamic configuration. Returns a JSON object with the dynamic configuration properties and values.
```
Comment (vtlim) [inline docs/api-reference/dynamic-configuration-api.md:43]:
```suggestion
Retrieves the current Coordinator dynamic configuration. Returns a JSON object with the dynamic configuration properties.
```

File: docs/api-reference/dynamic-configuration-api.md [hunk 2]
```diff
@@ -3,73 +3,663 @@ id: dynamic-configuration-api
...
+#### URL
+
+<code class="getAPI">GET</code> <code>/druid/coordinator/v1/config/history</code>
+
+#### Query parameters
+
+The endpoint supports a set of optional query parameters to filter results.
+
+* `interval`
+  * Type: String (ISO-8601)
```
Comment (vtlim) [inline docs/api-reference/dynamic-configuration-api.md:236]:
```suggestion
  * Type: String
```

File: docs/api-reference/dynamic-configuration-api.md [hunk 3]
```diff
@@ -3,73 +3,663 @@ id: dynamic-configuration-api
...
+  * Limit the results to the specified time interval. Delimited with `/`. For example, `2023-07-13/2023-07-19`. You can specify the default value of `interval` by setting `druid.audit.manager.auditHistoryMillis` in Coordinator `runtime.properties`. If not specified, `interval` defaults to one week.
```
Comment (vtlim) [inline docs/api-reference/dynamic-configuration-api.md:237]:
```suggestion
  * Limit the results to the specified time interval in ISO 8601 format delimited with `/`. For example, `2023-07-13/2023-07-19`. You can specify the default value of `interval` by setting `druid.audit.manager.auditHistoryMillis` in Coordinator `runtime.properties`. If not specified, `interval` defaults to one week.
```

File: docs/api-reference/dynamic-configuration-api.md [hunk 4]
```diff
@@ -3,73 +3,663 @@ id: dynamic-configuration-api
...
+
+<code class="postAPI">POST</code> <code>/druid/coordinator/v1/config</code>
+
+#### Header parameters
+
+The endpoint supports a set of optional header parameters to populate the `author` and `comment` fields in the configuration history.
+
+* `X-Druid-Author`
+  * Type: String
+  * A string representing the author making the configuration change.
```
Comment (vtlim) [inline docs/api-reference/dynamic-configuration-api.md:131]:
```suggestion
  * Author of the configuration change.
```

File: docs/api-reference/dynamic-configuration-api.md [hunk 5]
```diff
@@ -3,73 +3,663 @@ id: dynamic-configuration-api
...
+* `X-Druid-Comment`
+  * Type: String
+  * A string describing the update.
```
Comment (vtlim) [inline docs/api-reference/dynamic-configuration-api.md:134]:
```suggestion
  * Description for the update.
```

File: docs/api-reference/dynamic-configuration-api.md (hunk 3 above)
Comment (vtlim) [inline docs/api-reference/dynamic-configuration-api.md:237]: "To one week" -- which week is this? The most recent seven days?

File: docs/api-reference/dynamic-configuration-api.md [hunk 6]
```diff
@@ -3,73 +3,663 @@ id: dynamic-configuration-api
...
 ## Overlord dynamic configuration
 
-See [Overlord Dynamic Configuration](../configuration/index.md#overlord-dynamic-configuration) for details.
+The Overlord has dynamic configurations to tune how Druid assigns tasks to workers.
+For information on the supported properties, see [Overlord dynamic configuration](../configuration/index.md#overlord-dynamic-configuration).
+
+### Get dynamic configuration
+
+Retrieves the current Overlord dynamic configuration.
+Returns a JSON object with the dynamic configuration properties and values.
```
Comment (vtlim) [inline docs/api-reference/dynamic-configuration-api.md:313]:
```suggestion
Returns a JSON object with the dynamic configuration properties.
```

File: docs/api-reference/dynamic-configuration-api.md [hunk 7]
```diff
@@ -3,73 +3,663 @@ id: dynamic-configuration-api
...
+
+<code class="postAPI">POST</code><code>/druid/indexer/v1/worker</code>
+
+#### Header parameters
+
+The endpoint supports a set of optional header parameters to populate the `author` and `comment` fields in the configuration history.
+
+* `X-Druid-Author`
+  * Type: String
+  * A string representing the author making the configuration change.
```
Comment (vtlim) [inline docs/api-reference/dynamic-configuration-api.md:391]:
```suggestion
  * Author of the configuration change.
```

File: docs/api-reference/dynamic-configuration-api.md [hunk 8]
```diff
@@ -3,73 +3,663 @@ id: dynamic-configuration-api
...
+* `X-Druid-Comment`
+  * Type: String
+  * A string describing the update.
```
Comment (vtlim) [inline docs/api-reference/dynamic-configuration-api.md:394]:
```suggestion
  * Description for the update.
```

File: docs/api-reference/dynamic-configuration-api.md [hunk 9]
```diff
@@ -3,73 +3,663 @@ id: dynamic-configuration-api
...
+<code class="getAPI">GET</code> <code>/druid/indexer/v1/worker/history</code>
+
+
+#### Query parameters
+
+The endpoint supports a set of optional query parameters to filter results.
+
+* `interval`
+  * Type: String (ISO-8601)
+  * Limit the results to the specified time interval. Delimited with `/`. For example, `2023-07-13/2023-07-19`. You can specify the default value of `interval` by setting `druid.audit.manager.auditHistoryMillis` in Overlord `runtime.properties`. If not specified, `interval` defaults to one week.
```
Comment (vtlim) [inline docs/api-reference/dynamic-configuration-api.md:478]:
```suggestion
  * Type: String
  * Limit the results to the specified time interval in ISO 8601 format delimited with `/`. For example, `2023-07-13/2023-07-19`. You can specify the default value of `interval` by setting `druid.audit.manager.auditHistoryMillis` in Overlord `runtime.properties`. If not specified, `interval` defaults to one week.
```

File: docs/api-reference/dynamic-configuration-api.md [hunk 10]
```diff
@@ -3,73 +3,663 @@ id: dynamic-configuration-api
...
+        "blacklistedUntil": null
+    }
+]
+```
+
+</details>
+
+### Get scaling events
+
+Returns Overlord scaling events if auto-scaling runners are in use.
```
Comment (vtlim) [inline docs/api-reference/dynamic-configuration-api.md:619]: Should auto-scaling have a hyphen or not? See also how it's referred here: https://druid.apache.org/docs/latest/design/overlord/

**Review by vtlim — CHANGES_REQUESTED** [review]

File: docs/api-reference/dynamic-configuration-api.md (hunk 10 above)
Comment (ektravel) [reply docs/api-reference/dynamic-configuration-api.md:619]: "Autoscaling" appears to be more prevalent in Druid docs.

**Review by vtlim — APPROVED** [review]

**Pull request merged**